script.js

```diff
@@ -97,91 +107,155 @@ function highlightElements(elements, className) {
  */
 
 const checkIfWon = () => {
-    const isWin = foundPairs === wordPairs.length;
-    const isCurrentRoundOver = foundPairs % pairCount === 0;
+    const isWin = foundPairs === totalWordsInSessionCount;
+    const isCurrentRoundOver = foundPairs % pairsToRenderCount === 0;
     const pairsWereFound = foundPairs !== 0;
     const shouldSetupNextRound = isCurrentRoundOver && pairsWereFound && !isWin;
 
     if (shouldSetupNextRound) {
-      const lastSetOfPairsNumber = wordPairs.length % pairCount;
-      if (lastUsedPairIndex + pairCount > wordPairs.length) {
-        setupRound(wordPairs, lastUsedPairIndex + lastSetOfPairsNumber);
+      const lastSetOfPairsNumber = totalWordsInSessionCount % pairsToRenderCount;
+      if (lastUsedPairIndex + pairsToRenderCount > totalWordsInSessionCount) {
+        setupRound(shuffledKanjiHiraganaGlossaries, lastUsedPairIndex + lastSetOfPairsNumber);
       } else {
-        setupRound(wordPairs, lastUsedPairIndex + pairCount);
+        setupRound(shuffledKanjiHiraganaGlossaries, lastUsedPairIndex + pairsToRenderCount);
       }
     }
 
     if (isWin) {
-      // TODO: display the time it took to win the game
-      alert("You won!");
+      stopTimer();
     }
+};
+
+/**
+ * Timer functions
+ */
+
+const starTimer = () => {
+    gameStart = Date.now();
+    // Update timer every second
+    timerInterval = setInterval(updateTimer, 1000);
+};
+
+const stopTimer = () => {
+    // Stop the timer
+    clearInterval(timerInterval);
+    const gameEnd = Date.now();
+    // Convert to seconds
+    const gameDuration = (gameEnd - gameStart) / 1000;
+    alert(`Game duration: ${Math.floor(gameDuration)} seconds`);
+};
+
+const updateTimer = () => {
+    const currentTime = Date.now();
+    const elapsedTime = currentTime - gameStart;
+    const minutes = Math.floor(elapsedTime / 60000);
+    const seconds = Math.floor((elapsedTime % 60000) / 1000);
+    document.getElementById('timer').innerText = `${formatTime(minutes)}:${formatTime(seconds)}`;
+}
+/**
+ * 
+ * @param time - minutes or seconds as ints.
+ * @returns string formatted in mm:ss format.
+ */
+const formatTime = (time) => {
+    // If time is smaller than 10, add 0 in front, eg. 00:03, instead of 0:3.
+    return time < 10 ? `0${time}` : time;
 }
 
 window.addEventListener("load", () => {
-  // We are starting the game when the page is loaded - before that, we don't have the divs to work with (they are not rendered yet)
-  // Create the initial state of the game - generate the divs with words and kanjis in HTML.
-  setupRound(wordPairs, pairCount);
-  // TODO: add a function to start the timer here.
+  // We are starting the game when the page is loaded - before that, we don't have the divs to work with (they are not rendered yet).
+  // Create the initial state of the game - generate the divs with kanji and hiragana/katakana in HTML.
+  setupRound(shuffledKanjiHiraganaGlossaries, pairsToRenderCount);
+  starTimer();
```

Comment:
works nicely. Maybe next step would be to add a "start again" button for the whole game?